File: Acupay/Core Forms/PayStub/Entity/ThirteenthMonthPay.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("thirteenthmonthpay")>
    Public Class ThirteenthMonthPay

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        Public Property OrganizationID As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Created As Date

        Public Property CreatedBy As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        <Column("PayStubID")>
        Public Property PaystubID As Integer?

        Public Property BasicPay As Decimal

        Public Property Amount As Decimal

        <ForeignKey("PaystubID")>
        Public Overridable Property Paystub As Paystub

    End Class

End Namespace

File: AccuPay.Core/Entities/ThirteenthMonthPay.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Core.Entities
{
    [Table("thirteenthmonthpay")]
    public class ThirteenthMonthPay
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public virtual int? RowID { get; set; }

        public virtual int? OrganizationID { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public virtual DateTime Created { get; set; }

        public virtual int? CreatedBy { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public virtual DateTime? LastUpd { get; set; }

        public virtual int? LastUpdBy { get; set; }

        public virtual decimal BasicPay { get; set; }

        public virtual decimal Amount { get; set; }

        [Key]
        public int? PaystubID { get; set; }

        public virtual Paystub Paystub { get; set; }
    }
}